SimpleExec/Command.cs

```diff
@@ -49,14 +50,14 @@ public static string Read(string name, string args, string workingDirectory, boo
             }
         }
 
-        public static async Task<string> ReadAsync(string name, string args, string workingDirectory, bool noEcho)
+        public static async Task<string> ReadAsync(string name, string args, string workingDirectory, bool noEcho, CancellationToken cancellationToken)
         {
             using (var process = new Process())
             {
                 process.StartInfo = ProcessStartInfo.Create(name, args, workingDirectory, true);
-                await process.RunAsync(noEcho).ConfigureAwait(false);
+                await process.RunAsync(noEcho, cancellationToken).ConfigureAwait(false);
 
-                if (process.ExitCode != 0)
+                if (process.ExitCode != 0 && !cancellationToken.IsCancellationRequested)

```

Comment:
I believe this has the same `NullReferenceException` problem as in `RunAsync()`.